lib/chef/provider/profile.rb

```diff
@@ -0,0 +1,258 @@
+          config_uuid(@new_profile_hash) if @new_profile_hash
+
+        if @new_profile_hash
+          @new_profile_identifier = @new_profile_hash['PayloadIdentifier'] ||
+            @new_resource.identifier ||
+            @new_resource.profile_name
+        else
+          @new_profile_identifier = @new_resource.identifier ||
+          @new_resource.profile_name
+        end
+
+        current_profile = all_profiles['_computerlevel'].find {
+          |item| item['ProfileIdentifier'] ==
+             @new_profile_identifier
+        }
+        @current_resource.profile(current_profile)
+
+        @current_resource.installed(profile_installed?)
+      end
+
+      def define_resource_requirements
+        requirements.assert(:remove) do |a|
+          if @new_profile_identifier
+            a.assertion {
+              !@new_profile_identifier.nil? and
+              !@new_profile_identifier.end_with?('.mobileconfig') and
+              /^\w+(?:\.\w+)+$/.match(@new_profile_identifier)
+            }
+            a.failure_message RuntimeError, "when removing using the identifier attribute, it must match the profile identifier"
+          else
+          new_profile_name = @new_resource.profile_name
+            a.assertion {
+              !new_profile_name.end_with?('.mobileconfig') and
+              /^\w+(?:\.\w+)+$/.match(new_profile_name)
+            }
+            a.failure_message RuntimeError, "When removing by resource name, it must match the profile identifier "
+          end
+        end
+
+        requirements.assert(:install) do |a|
+          if @new_profile_hash.is_a?(Hash)
+            a.assertion {
+              @new_profile_hash.include?('PayloadIdentifier')
+            }
+            a.failure_message RuntimeError, "The specified profile does not seem to be valid"
+          end
+        end
+      end
+
+      def action_install
+        unless @current_resource.installed
+          converge_by("install profile #{@new_profile_hash['PayloadIdentifier']}") do
+            profile_path = write_profile_to_disk
+            install_profile(profile_path)
+            get_installed_profiles(update=true)
+          end
+        end
+      end
+
+      def action_remove
+        # Clean up profile after removing it
+        if @current_resource.installed
+          converge_by("remove profile #{@new_resource.profile_name}") do
+            remove_profile
+            get_installed_profiles(update=true)
+          end
+        end
+      end
+
+      def load_profile_hash(new_profile)
+          # file must exist in cookbook
+          unless new_profile.end_with?('.mobileconfig')
+            error_string = "'#{new_profile}' is not a valid profile"
+            raise RuntimeError, error_string
+          end
+          unless cookbook_file_available?(new_profile)
+            error_string = "'#{new_profile}' not found in cookbook"
+            raise Chef::Exceptions::FileNotFound, error_string
```

Comment:
that is, you wouldn't really know where it came from. unless your profile name was "my_profile" you might not realize it had anything to do with the profile provider.